ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/HD/DiffSeq.hs

```diff
@@ -0,0 +1,40 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Ouroboros.Storage.LedgerDB.HD.DiffSeq (tests) where
+
+import           Data.Proxy
+
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+
+import           Data.Semigroupoid.Auto
+import           Data.Semigroupoid.Laws
+
+import           Test.Util.Orphans.Arbitrary ()
+import           Test.Util.Orphans.DiffSeq.Arbitrary ()
+
+import           Ouroboros.Consensus.Storage.LedgerDB.HD.DiffSeq
+
+
+
+tests :: TestTree
+tests = testGroup "DiffSeq" [
+    testGroupWithProxy (Proxy @(RootMeasure X X)) [
+        testSemigroupLaws
+      , testMonoidLaws
+      , testGroupLaws
+      ]
+  , testGroupWithProxy (Proxy @(Auto (RootMeasure X X))) [
+        testSemigroupoidLaws
+      , testGroupoidLaws
+      ]
+  , testGroupWithProxy (Proxy @(InternalMeasure X X)) [
+        testSemigroupLaws
+      , testMonoidLaws
+      ]
+  , testGroupWithProxy (Proxy @(Auto (InternalMeasure X X))) [
+        testSemigroupoidLaws
+      ]
+  ]
+
+type X = Small Int
```

Comment:
What if we omit the type synonim? Maybe it's better to remove this level of indirection. Besides `X` is not very descriptive :)